bundle/src/main/java/com/adobe/acs/commons/cloudservices/pwa/impl/PwaManifestServlet.java

```diff
@@ -0,0 +1,167 @@
+package com.adobe.acs.commons.cloudservices.pwa.impl;
+
+import com.adobe.acs.commons.cloudservices.pwa.Configuration;
+import com.day.cq.commons.jcr.JcrConstants;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.*;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.sling.commons.mime.MimeTypeService;
+import org.apache.sling.models.factory.ModelFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.stream.StreamSupport;
+
+import static com.adobe.acs.commons.cloudservices.pwa.impl.Constants.*;
+
+@Component(
+        service = Servlet.class,
+        property = {
+                "sling.servlet.resourceTypes=cq:Page",
+                "sling.servlet.methods=" + HttpConstants.METHOD_GET,
+                "sling.servlet.selectors=pwa.manifest",
+                "sling.servlet.extensions=json",
+                "sling.servlet.extensions=webmanifest"
+
+        }
+)
+public class PwaManifestServlet extends SlingSafeMethodsServlet {
+    private static final Logger log = LoggerFactory.getLogger(PwaManifestServlet.class);
+
+    @Reference
+    private MimeTypeService mimeTypeService;
+
+    @Reference
+    private ModelFactory modelFactory;
+
+
+    @Reference
+    private ResourceResolverFactory resourceResolverFactory;
+
+
+    @Override
+    protected final void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws ServletException, IOException {
+        response.setContentType("application/x-web-app-manifest+json; charset=UTF-8");
+        response.setCharacterEncoding("UTF-8");
+
+
+        ResourceResolver serviceResourceResolver = null;
+        try {
+            serviceResourceResolver = resourceResolverFactory.getServiceResourceResolver(AUTH_INFO);
```

Comment:
Why are we using service users in the context of a Request? The user's permissions should be used, and ACLs should be applied accordingly (like we do for Templates, etc.) Or, CUGs could be set on the resource so when its replicated, they are accessible as needed.

This should only be needed on Publish as well.

Now, anyone that hits this URL can access the payload (ie. if this is a secured site, this URL is not secure anymore).